app/controllers/registrations_controller.rb

```diff
@@ -42,5 +46,89 @@ def update
 # check if we need the current password to update fields
 def needs_password?(member, params)
   params[:member][:password].present? ||
-  params[:member][:password_confirmation].present?
+  params[:member][:password_confirmation].present? ||
+  params[:member][:deleted].present?
+end
+
+# marking someone as deleted has various effects
+def markdelete
+
+    # move any of their crops to cropbot
+    if Role.crop_wranglers && (Role.crop_wranglers.include? @member)
+      cropbot = Member.find_by_login_name('ex_wrangler')
+      if Crop.find_by(creator: @member)
+        # this is ugly, need to make it more efficient
+        Crop.where(creator: @member).each do |crop|
+          Crop.update(crop, creator: cropbot)
+          crop.save!
+        end
+      end
+    end
+
+    # mark their comments as deleted
+    ex_member = Member.find_by_login_name('ex_member')
+    if Comment.find_by(author: @member)
+      Comment.where(author: @member).each do |comment|
+        Comment.update(comment, author: ex_member, body: "This comment was removed as the author deleted their account.")
```

Comment:
I'm not convinced this is the best approach. It preserves threading, but it's still destructive - we can't undo the deletion without restoring from backup. Alternatively, we could make comments/posts/gardens/harvests/etc soft-deletable, and render soft-deleted comments as "This comment was removed...". That would allow us to translate the "this comment was removed..." message. On the other hand:

 - it would be more work
 - it would still be destructive: delete a comment, then delete your account. Then undelete your account. Should we undelete the comment?

Maybe we could scope all gardens/comments/etc to "owner has not been soft-deleted"? I dunno. I'm tired.